examples/customer-api/app/routes/_index.tsx

```diff
@@ -12,11 +12,26 @@ export async function loader({context}: LoaderArgs) {
       }
       `);
 
-    return json({
-      user,
-    });
+    return json(
+      {
+        user,
+      },
+      {
+        headers: {
+          'Set-Cookie': await context.session.commit(),
+        },
+      },
```

Comment:
This is the biggest thing that sucks about the abstraction at the moment. _Any loader_ that calls the customer account api client _must_ return the committed session. The only alternative I can think of is to always commit the session inside `server.ts`.

Reply:
Someone brought up a big downside of doing so. I'm trying to remember what the concern was. 🤔

Reply:
Could we change `HydrogenSession` to actually track changes somehow so we know if it's "dirty"? 

Something like this:

```js
class HydrogenSession {
// ...

  #isDirty = false;

  get isDirty() {
    return this.#isDirty;
  }

  set(...args) {
    this.#session.set(...args);
    this.#isDirty = true;
  }

  async commit() {
    const result = await this.#sessionStorage.commitSession(this.#session);
    this.#isDirty = false;
    return result;
  }

}
```

Then check `session.isDirty` before committing in `server.ts` / remix-oxygen? 🤔

Reply:
I implemented this out, and after a discussion with @juanpprieto, I don't think I feel good about it for now. I'd rather just require the user to manually commit the session for now. Though, if I can convince the API folks to differentiate between these two scenarios, I think I might be able to warn the user they didn't persist/commit their session:

1. Try to refresh the auth token, but the refresh token is expired
1. Try to refresh the auth token, but the refresh token has already been used